shell/platform/linux/public/flutter_linux/fl_platform_views.h

```diff
@@ -0,0 +1,137 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_PLATFORM_VIEWS_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_PLATFORM_VIEWS_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <gtk/gtk.h>
+
+#include "fl_message_codec.h"
+#include "fl_value.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlPlatformView,
+                         fl_platform_view,
+                         FL,
+                         PLATFORM_VIEW,
+                         GObject)
+
+/**
+ * FlPlatformView:
+ *
+ * #FlPlatformView is an abstract class that wraps a #GtkWidget for embedding in
+ * the Flutter hierarchy.
+ */
+
+struct _FlPlatformViewClass {
+  GObjectClass parent_class;
+
+  /**
+   * Virtual method called when Flutter needs the #GtkWidget for embedding.
+   * @platform_view: an #FlPlatformView.
+   *
+   * Returns: (allow-none): an #GtkWidget or %NULL if no widgets can be
+   * provided.
+   */
+  GtkWidget* (*get_view)(FlPlatformView* platform_view);
+};
+
+/**
+ * fl_platform_view_get_view:
+ * @platform_view: an #FlPlatformView.
+ *
+ * Returns: (allow-none): an #GtkWidget or %NULL if no widgets can be provided.
+ */
+GtkWidget* fl_platform_view_get_view(FlPlatformView* platform_view);
+
+//
+
+G_DECLARE_INTERFACE(FlPlatformViewFactory,
+                    fl_platform_view_factory,
+                    FL,
+                    PLATFORM_VIEW_FACTORY,
+                    GObject)
+
+/**
+ * FlPlatformViewFactory:
+ *
+ * #FlPlatformViewFactory vends #FlPlatformView objects for embedding.
```

Comment:
It might be good to document which component takes care of life-cycle of the created platform view along with a high level overview of how one is supposed to use the methods on this struct, for example something about message codec and views.

Reply:
Done.